src/Optimization/hiopAlgFilterIPM.hpp

```diff
@@ -339,8 +345,74 @@ class hiopAlgFilterIPMQuasiNewton : public hiopAlgFilterIPMBase
   virtual ~hiopAlgFilterIPMQuasiNewton();
 
   virtual hiopSolveStatus run();
+
+  // note that checkpointing is only available with a axom-enabled build
+#ifdef HIOP_USE_AXOM
+  /**
+   * @brief Save state of HiOp algorithm to a sidre data store.
+   * @param data_store a pointer to DataStore
+   *
+   * @details 
+   * A new sidre::group "hiop solver" is created within data_store. Then a sidre::View 
+   * is created within the group for each of the algorithm's states and the states are
+   * copied in the corresponding views. 
+   */
+  virtual void save_state_to_data_store(::axom::sidre::DataStore* data_store);
```

Comment:
```suggestion
  virtual void save_state_to_data_store(::axom::sidre::DataStore &data_store);
```
I think a reference makes the meaning clearer here. `data_store` is not allowed to be a nullptr, and the client (caller) is *not* transferring ownership of the pointer to `save_state_to_data_store`.

Reply:
100% agree, will use refs.